src/managers/config.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling configuration building + writing."""
+
+import logging
+
+from ops.model import ConfigData
+
+from core.cluster import ClusterState
+from core.workload import WorkloadBase
+from literals import CONFIG_FILE
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PROPERTIES = """
```

Comment:
I would like to avoid loading the config-file to avoid anyone overwriting config (as explained in the config "vision"/discussion some days ago). We could argue about moving this to a different place in the code or a different format (currently it's in line with other charms doing it the same way), but I think the charm should be able to fully control the config and not rely on anything written on disk.

Reply:
Thanks for the advice! I'd like to have it stored as part of the code, meaning I'll move the plain text to a yaml config file and load it from there. I will make this a separate PR if that's fine?